backbone.js

```diff
@@ -1526,11 +1555,36 @@
     // treated as `null` to normalize cross-browser behavior.
     _extractParameters: function(route, fragment) {
       var params = route.exec(fragment).slice(1);
-      return _.map(params, function(param, i) {
+      var decodedParams = _.map(params, function(param, i) {
         // Don't decode the search params.
         if (i === params.length - 1) return param || null;
         return param ? decodeURIComponent(param) : null;
       });
+
+      if (this.useParamsObject) {
+        // get routeInfo, which has array of param names
+        var routeInfo = this._routeInfos[route.toString()];
+
+        var routeData = {
+          name: routeInfo.name
```

Comment:
I am currently using that right now in my app, where I hold on to current route data. And I use route.name to know what route it is.

Something along the line of `router.getCurrentRoute`, since it's not part of the route handler, `name` is what tells me what current route is.